datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1157,10 +1157,8 @@ impl LogicalPlan {
         Ok(if let LogicalPlan::Prepare(prepare_lp) = plan_with_values {
             param_values.verify(&prepare_lp.data_types)?;
             // try and take ownership of the input if is not shared, clone otherwise
-            match Arc::try_unwrap(prepare_lp.input) {
-                Ok(input) => input,
-                Err(arc_input) => arc_input.as_ref().clone(),
-            }
+            Arc::try_unwrap(prepare_lp.input)
+                .unwrap_or_else(|arc_input| arc_input.as_ref().clone())
```

Comment:
We can use [Arc::unwrap_or_clone](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.unwrap_or_clone) to simplify it.

Reply:
good point, done!